src/nplinker/scoring/linking/data_linking.py

```diff
@@ -108,7 +108,18 @@ def find_correlations(self, include_singletons=False):
         self.data_family_mapping(include_singletons=include_singletons)
         self.correlation_matrices(type='fam-gcf')
 
-    def collect_mappings_spec(self, spectra):
+    def collect_mappings_spec(self, obj: Iterable[Spectrum]|Iterable[MolecularFamily]):
+        if isinstance(next(obj), Spectrum):
+            mapping_spec = self._collect_mappings_from_spectra(obj)
+        elif isinstance(next(obj), MolecularFamily):
+            mapping_spec = self._collect_mappings_from_molecular_families(obj)
```

Comment:
```suggestion
    def collect_mappings_spec(self, obj: Sequence[Spectrum]|Sequence[MolecularFamily]):
        if isinstance(obj[0], Spectrum):
            mapping_spec = self._collect_mappings_from_spectra(obj)
        elif isinstance(obj[0], MolecularFamily):
            mapping_spec = self._collect_mappings_from_molecular_families(obj)
```
`next(Iterable)` does not work. `next` works on iterator.
 
Don't forget to add `from typing import Sequence`.